lib/xcodeproj/project/object/build_configuration.rb

```diff
@@ -6,6 +6,8 @@ module Object
       # {PBXProject} or a {PBXNativeTarget}.
       #
       class XCBuildConfiguration < AbstractObject
+        MUTUAL_RECURSION_SENTINEL = 'xcodeproj.mutual_recursion_sentinel'.freeze
```

Comment:
Might actually be more performant to make this `= Object.new.freeze`. Also, I think it can be a `private_constant`?

Reply:
1. It doesn't work with `Object.new.freeze` (actually `::Object.new`), because `String` specific methods are called on it, and so it crashes.
2. TIL about `private_constant`.